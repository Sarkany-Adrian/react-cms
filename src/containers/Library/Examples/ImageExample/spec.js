import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImageExample from 'containers/Library/Examples/ImageExample';

describe('ImageExample', () => {
  it('Should render all ImageExample components', () => {
    const wrapper = shallow(<ImageExample />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('check if image onClick method is called', () => {
    const onClick = jest.fn();
    const wrapper = mount(<ImageExample onClick={onClick} />);
    const clickHandler = wrapper.find('img').at(6);
    clickHandler.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
