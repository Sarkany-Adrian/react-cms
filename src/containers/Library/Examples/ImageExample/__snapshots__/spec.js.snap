// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageExample Should render all ImageExample components 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Image Atom:
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        Image:
      </p>
      <Image
        alt="JSNinja"
        circle={false}
        crisp={false}
        errorImage="error_image.png"
        height="100"
        lg={false}
        md={false}
        rounded={false}
        sm={false}
        src="js_ninja.png"
        xs={false}
      />
    </div>
    <div
      className="col"
    >
      <p>
        Image with error:
      </p>
      <Image
        alt="Image"
        circle={false}
        crisp={false}
        errorImage="error_image.png"
        height="100"
        lg={false}
        md={false}
        rounded={false}
        sm={false}
        src=""
        xs={false}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        ImageÂ 
        <code>
          circle height=
          45
           width=
          45
        </code>
        :
      </p>
      <Image
        alt="test"
        circle={true}
        crisp={false}
        errorImage="error_image.png"
        height={45}
        lg={false}
        md={false}
        rounded={false}
        sm={false}
        src="placeimg.jpg"
        width={45}
        xs={false}
      />
    </div>
    <div
      className="col"
    >
      <p>
        Image 
        <code>
          rounded sm
        </code>
        :
      </p>
      <Image
        alt="test"
        circle={false}
        crisp={false}
        errorImage="error_image.png"
        lg={false}
        md={false}
        rounded={true}
        sm={true}
        src="placeimg.jpg"
        xs={false}
      />
    </div>
    <div
      className="col"
    >
      <p>
        Image 
        <code>
          crisp md
        </code>
        :
      </p>
      <Image
        alt="test"
        circle={false}
        crisp={true}
        errorImage="error_image.png"
        lg={false}
        md={true}
        rounded={false}
        sm={false}
        src="placeimg.jpg"
        xs={false}
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <p>
          Image 
          <code>
            default lg
          </code>
          :
        </p>
        <Image
          alt="test"
          circle={false}
          crisp={false}
          errorImage="error_image.png"
          lg={true}
          md={false}
          rounded={false}
          sm={false}
          src="placeimg.jpg"
          xs={false}
        />
      </div>
      <div
        className="col"
      >
        <p>
          Image 
          <code>
            rounded responsive onClick
          </code>
          :
        </p>
        <Image
          alt="test"
          circle={false}
          className="onClick-test"
          crisp={false}
          errorImage="error_image.png"
          lg={false}
          md={false}
          responsive={true}
          rounded={true}
          sm={false}
          src="placeimg.jpg"
          xs={false}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
